crawl-ref/source/stairs.cc

```diff
@@ -627,23 +635,13 @@ void floor_transition(dungeon_feature_type how,
         how = branches[old_level.branch].entry_stairs;
     }
 
-    // Special messages on returning from portal vaults, Abyss, Pan, etc.
-    if (!is_connected_branch(old_level.branch)
-        && player_in_connected_branch()
-        && old_level.branch != you.where_are_you)
-    {
-        mprf("Welcome %sto %s!",
-             you.chapter == CHAPTER_POCKET_ABYSS ? "" : "back ",
-             branches[you.where_are_you].longname);
-    }
-
-    // Falling down the stairs or portal.
+    // Check for falling down the stairs or portal. (Why can't you fall in the abyss?)
```

Comment:
3f120941 / https://crawl.develz.org/mantis/view.php?id=6975 suggests that's the reason; 0a4efdf8 is also involved, but that predates useful commit messages.